cmd2/parsing.py

```diff
@@ -0,0 +1,387 @@
+#
+# -*- coding: utf-8 -*-
+"""Statement parsing classes for cmd2"""
+
+import re
+import shlex
+from typing import List, Tuple
+
+import cmd2
+
+BLANK_LINE = '\n'
+
+def _comment_replacer(match):
+    s = match.group(0)
+    if s.startswith('/'):
+        # treat the removed comment as an empty string
+        return ''
+    else:
+        return s
+
+class Statement(str):
+    """String subclass with additional attributes to store the results of parsing.
+    
+    The cmd module in the standard library passes commands around as a
+    string. To retain backwards compatibility, cmd2 does the same. However, we
+    need a place to capture the additional output of the command parsing, so we add
+    our own attributes to this subclass.
+
+    The string portion of the class contains the arguments, but not the command, nor
+    the output redirection clauses.
+    """
+    def __init__(self, object):
+        self.raw = str(object)
+        self.command = None
+        self.multilineCommand = None
```

Comment:
Recommend using PEP8-compliant "snake_case" for member variable names for new and/or significantly modified code.  ``cmd2`` didn't originally comply with this.  However, moving in this direction makes us play more nicely with tools like pylint.

This comment applies to a number of member variable names.

Reply:
I had preserved the camel case in an effort to improve backward compatibility.  However, since anyone who is using the pyparsing objects in their own code will have to fix it anyway, this is the best (and maybe only) time to move it to snake case. Fixed and pushed to `ply` branch.